lib/streamlit/ReportThread.py

```diff
@@ -23,15 +23,53 @@
 ReportContext = namedtuple(
     "ReportContext",
     [
-        # The main DeltaGenerator for the report
+        # (DeltaGenerator) The main DeltaGenerator for the report
         "main_dg",
-        # The sidebar DeltaGenerator for the report
+        # (DeltaGenerator) The sidebar DeltaGenerator for the report
         "sidebar_dg",
-        # The Widgets state object for the report
+        # (Widgets) The Widgets state object for the report
         "widgets",
+        # (_WidgetIDSet) The set of widget IDs that have been assigned in the
+        # current report run. This set is cleared at the start of each run.
+        "widget_ids_this_run",
     ],
 )
 
+
+class _WidgetIDSet(object):
+    """Stores a set of widget IDs. Safe to mutate from multiple threads."""
```

Comment:
I have been told to use
```
"""Stores a set of widget IDs. Safe to mutate from multiple threads.
"""
```

See the newline.

Reply:
My understanding is that this rule only applies when the docstring is multiline. For singleline docstrings, the closing triple-quote goes on the same line, per https://www.python.org/dev/peps/pep-0257/#one-line-docstrings